packages/@atjson/source-html/src/converter/social-embeds.ts

```diff
@@ -0,0 +1,124 @@
+import Document, { Annotation } from "@atjson/document";
+import OffsetSource, { IframeEmbed } from "@atjson/offset-annotations";
+import UrlSource from "@atjson/source-url";
+
+function isInstagramOrTwitterBlockquote(annotation: Annotation<any>) {
+  if (annotation.type !== "blockquote") {
+    return false;
+  }
+
+  let classes = annotation.attributes.class;
+
+  if (!classes) {
+    return false;
+  }
+
+  let classList = classes.split(" ");
+  return (
+    classList.includes("instagram-media") || classList.includes("twitter-tweet")
+  );
+}
+
+function isFacebookDiv(annotation: Annotation<any>) {
+  if (annotation.type !== "div") {
+    return false;
+  }
+
+  return (
+    annotation.attributes.dataset &&
+    annotation.attributes.dataset.href &&
+    new URL(annotation.attributes.dataset.href).host === "www.facebook.com"
+  );
+}
+
+export default function(doc: Document) {
+  doc.where({ type: "-html-iframe" }).update(iframe => {
+    let { height, width, src: url } = iframe.attributes;
+    let urlDoc = UrlSource.fromRaw(url).convertTo(OffsetSource);
```

Comment:
This is very clever! I really like this reuse— are there ways that we can prevent requiring `@atjson/source-url` from creating a dependency graph that results in more packages than we want being installed? This doesn't need to be answered in this PR, but I feel like we've been continuing to encounter these cases (should we make all the core packages peers?)

Reply:
Extracting this logic to util functions in offset-annotations was a good suggestion! I think not having a dependency on source-url to be a major plus.